### ISSUE TITLE
Fix modules that need to restart stuff and graceful reload

### ISSUE THREAD

**Issue by miekg** [coredns/coredns#140]
Things like prometheus need to close and rebind to their socket after a graceful reload, there is no hook for that in CoreDNS.

Think best is just to call Shutdown and re-initialize.

**Comment by miekg** [comment]
Maybe call shutdown with a boolean indicating a restart?

**Comment by miekg** [comment]
Needs to happen for:
- health
- metrics (prometheus)
- pprof

**Comment by miekg** [comment]
Starting and reloading

```
Activating privacy features...
atoom.net.:1053
Warning: File descriptor limit 1024 is too low for production sites. At least 4096 is recommended. Set with "ulimit -n 4096".
2016/04/29 07:18:30 [INFO] SIGUSR1: Reloading
2016/04/29 07:18:30 [INFO] Restarting
```

This works for the above middleware.

**Comment by miekg** [comment]
Fix with #142
